include/FactoryPieces.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <memory> 
+#include <map>
+#include "Piece.h"
+using std::unique_ptr;
+
+class FactoryPieces
+{
+public:
+	using pFunc = unique_ptr<Piece>(*)(const char&, const Location&, PlayerColor color, Board* board);

```

Comment:
pFnc isn't a good name for this function pointer type. Try to think of something more readable that will make the reader understand its purpose. Maybe "creation_func"?